sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -273,29 +274,37 @@ abstract class TypeCoercionBase {
     @tailrec private def getWidestTypes(
         children: Seq[LogicalPlan],
         attrIndex: Int,
-        castedTypes: mutable.Queue[DataType]): Seq[DataType] = {
+        castedTypes: mutable.Queue[Option[DataType]]): Seq[Option[DataType]] = {
       // Return the result after the widen data types have been found for all the children
       if (attrIndex >= children.head.output.length) return castedTypes.toSeq
 
       // For the attrIndex-th attribute, find the widest type
       findWiderCommonType(children.map(_.output(attrIndex).dataType)) match {
         // If unable to find an appropriate widen type for this column, return an empty Seq
-        case None => Seq.empty[DataType]
+        case None =>
+          castedTypes.enqueue(None)
+          getWidestTypes(children, attrIndex + 1, castedTypes)
         // Otherwise, record the result in the queue and find the type for the next column
         case Some(widenType) =>
-          castedTypes.enqueue(widenType)
+          castedTypes.enqueue(Some(widenType))
           getWidestTypes(children, attrIndex + 1, castedTypes)
```

Comment:
the code can be simplified
```
findWiderCommonType(children.map(_.output(attrIndex).dataType)).map { widenTypeOpt =>
  castedTypes.enqueue(widenTypeOpt)
  getWidestTypes(children, attrIndex + 1, castedTypes)
}
```

Reply:
`findWiderCommonType` returns `Opion[DataType]`. `map` can iterate over the `DataType` if any, but we still need to enqueue the `None`.

Reply:
Just simplified to

```scala
val widenTypeOpt = findWiderCommonType(children.map(_.output(attrIndex).dataType))
castedTypes.enqueue(widenTypeOpt)
getWidestTypes(children, attrIndex + 1, castedTypes)
```